library/alloc/src/vec/spec_from_elem.rs

```diff
@@ -17,6 +18,18 @@ impl<T: Clone> SpecFromElem for T {
     }
 }
 
+impl<T: Clone + IsZero> SpecFromElem for T {
+    #[inline]
+    default fn from_elem<A: Allocator>(elem: T, n: usize, alloc: A) -> Vec<T, A> {
+        if elem.is_zero() {
+            return Vec { buf: RawVec::with_capacity_zeroed_in(n, alloc), len: n };
+        }
+        let mut v = Vec::with_capacity_in(n, alloc);
+        v.extend_with(n, ExtendElement(elem));
+        v
+    }
+}
+
 impl SpecFromElem for i8 {
```

Comment:
Are this impls is really necessary? Seems that LLVM successfully replaces `vec![true; n]` with memset, it would happen for u8/i8 too.